src/main/scala/TypeClass.scala

```diff
@@ -0,0 +1,49 @@
+trait CanAdd[A] {
+  def plus(a: A, b: A): A
+  def apply(a: A, b: A): A = plus(a, b)
```

Comment:
let's remove the apply, I think .plus() is much easier to read anyway

Reply:
done